packages/cell-toolbar-extension/src/index.ts

```diff
@@ -40,10 +59,21 @@ const cellToolbar: JupyterFrontEndPlugin<void> = {
           )
         : undefined;
 
-    app.docRegistry.addWidgetExtension(
-      'Notebook',
-      new CellBarExtension(app.commands, toolbarItems)
-    );
+    const extension = new CellBarExtension(app.commands, toolbarItems);
+
+    // Wait for the application to be restored and
+    // for the settings for this plugin to be loaded
+    if (settingRegistry !== null) {
+      void settingRegistry.load(PLUGIN_ID).then(setting => {
+        // Read the settings
+        loadSetting(setting);
+
+        // Listen for your plugin setting changes using Signal
+        setting.changed.connect(loadSetting);
+      });
```

Comment:
There is a tradeoff here (loading immediately if enabled, vs potentially flashing when disabled), so I will not insist on this, but in case if you did not consider it previously I wanted to suggest:

```suggestion
      const settings = await settingRegistry.load(PLUGIN_ID);

      // Read the settings
      loadSetting(setting);

      // Listen for your plugin setting changes using Signal
      setting.changed.connect(loadSetting);
```